### DISCUSSION TITLE
PHPStorm / IntelliJ editor extension

### DISCUSSION THREAD

**Discussion by GreenImp** [peggyjs/peggy#327] [General]
Hi,
Is there any chance of someone developing an editor extension for PHPStorm / IntelliJ IDEs? I can see that there are ones for various other IDES here: https://github.com/peggyjs/peggy/wiki/Editor-Extensions

I've found this one: ~https://github.com/peggyjs/code-peggy-language~ https://github.com/qlonik/intellij-peggy/ but it doesn't support the latest versions of PHPStorm and, from what I can tell, it doesn't actually provide syntax highlighting, which is my the biggest requirement.

I would consider trying myself, but I'm honestly not sure where to start. I've glanced over the docs, and I think it's a bit out of my skill set: https://plugins.jetbrains.com/docs/intellij/custom-language-support.html

I'm aware that most / all of these seem to have been built by third party developers, and there's no obligation for anyone to build this stuff, but I thought it couldn't hurt to ask.

**Comment by hildjj** [comment]
code-peggy-language does provide syntax highlighting, at least on my setup (VS Code on Mac). 

I'm not planning on buying a license to either of those platforms.  If someone else wants to build something, we will point to it so others can find it.

**Reply by GreenImp** [reply]
Sorry @hildjj I put the wrong link on there! :facepalm:
I meant this one: https://github.com/qlonik/intellij-peggy/ which is for IntelliJ IDEs

That's perfectly understandable about not wanting to purchase a license. It might be worth pointing out that they provide free licenses to OS projects: https://www.jetbrains.com/community/opensource/#support

However, I also understand that you simply may not want to build a plugin for it, as it's something else to support.
